src/order/category/DecLinOrder.lean

```diff
@@ -0,0 +1,87 @@
+/-
+Copyright (c) 2019 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+
+import category_theory.concrete_category
+import order.basic
+
+/-!
+# Category instances for ordered structures
+
+We introduce
+* Preorder     : the category of preorders and monotone functions.
+* PartialOrder : the category of partial orders and monotone functions.
+* LinOrder     : the category of linear orders and monotone functions.
+* DecLinOrder  : the category of decidable linear orders and monotone functions.
+
+and the appropriate forgetful functors between them.
+-/
+
+
+universes u v
+
+open category_theory
+
+/-- The category of preorders and monotone maps. -/
+@[reducible]
```

Comment:
Are the categories defined in this file intended to be permanently `[reducible]`? I'm not really sure what the ramifications are here, but this PR does go to the trouble of making a bunch of other categories not `[reducible]`...

Reply:
No, this was just an oversight. After the discussion of Zulip I’m now second guessing this whole `local attribute [reducible]` business. In particular I’m worried that it may be contributing to the problems in `CommRing/adjunction.lean`.